src/Workspaces/Core/Portable/CodeGeneration/CodeGenerationHelpers.cs

```diff
@@ -177,9 +177,18 @@ public static IEnumerable<ISymbol> GetMembers(INamedTypeSymbol namedType)
         {
             Contract.ThrowIfNull(symbol);
 
-            return options != null && options.ReuseSyntax && symbol.DeclaringSyntaxReferences.Length == 1
-                ? symbol.DeclaringSyntaxReferences[0].GetSyntax() as T
-                : null;
+            if (options.ReuseSyntax && symbol.DeclaringSyntaxReferences.Length == 1)
+            {
+                var reusableNode = symbol.DeclaringSyntaxReferences[0].GetSyntax();
+                if (reusableNode is T)
+                {
+                    var leadingTrivia = reusableNode.GetLeadingTrivia().Where(trivia => !trivia.IsDirective);
+                    var trailingTrivia = reusableNode.GetTrailingTrivia().Where(trivia => !trivia.IsDirective);
```

Comment:
directives can't be in trailing trivia.

Reply:
Fixed. Also I found some caller of this function is trying to use the Parent of the return value here. So we can't easily change the syntax node here.  And I move all these logic to the caller